gems/aws-sdk-core/lib/aws-sdk-core/xml/error_handler.rb

```diff
@@ -48,16 +47,34 @@ def error_data(context, code)
             # match modeled shape name
             error_shape_code = rule.shape['error']['code'] if rule.shape['error']
             match = (code == error_shape_code || code == rule.shape.name)
-            if match && rule.shape.members.any?
-              data = Parser.new(rule).parse(context.http_response.body_contents)
-            end
+            next unless match && rule.shape.members.any?
+
+            data = parse_error_data(rule, context.http_response.body_contents)
```

Comment:
I know I made this change, but I'm still slightly worried about the change.  My understanding of the parsing of XML errors is that in all protocols they MUST have the `<Errors>` or `<Errors>` tags - and so this old code would NEVER have worked.  I think this ended up being fine because these services don't actually model many errors?

Reply:
Per our offline conversation, I'm going to scan through our existing services to see if there's any existence of a modeled error response (this specific error handler is used for query, ec2 and rest-xml protocols so not many services to review) and also make some calls to see how things are being parsed if defined.

Reply:
I did some digging on how error responses are coming back so posting here for historic context:

---
**Service**: S3 (rest-xml protocol)
**Operation**: `GetObject` ([model](https://github.com/aws/aws-sdk-ruby/blob/431c33756ae4c3a020a9aa26e3fce9ae0f8673d6/apis/s3/2006-03-01/api-2.json#L589-L611))
**Error Shape**: `InvalidObjectState` ([model](https://github.com/aws/aws-sdk-ruby/blob/431c33756ae4c3a020a9aa26e3fce9ae0f8673d6/apis/s3/2006-03-01/api-2.json#L4953-L4961))
* We received the following error response:
```
-> "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>InvalidObjectState</Code><Message>The operation is not valid for the object's storage class</Message><StorageClass>DEEP_ARCHIVE</StorageClass><RequestId>fooID</RequestId><HostId>fooHostId</HostId></Error>"
```
* I was able to inspect the modeled error response like so:
```
(byebug) err.inspect
"#<Aws::S3::Errors::InvalidObjectState: The operation is not valid for the object's storage class>"
(byebug) err.class
Aws::S3::Errors::InvalidObjectState
(byebug) err.message
"The operation is not valid for the object's storage class"
(byebug) err.storage_class
"DEEP_ARCHIVE"
```

As a sanity check, I also looked at custom error code, which is very specific to query protocol: https://smithy.io/2.0/aws/protocols/aws-query-protocol.html#awsquery-error-response-code

**Service**: CloudWatch (query protocol, using the xml's error handler)
**Operation**: `GetDashboard` ([model def](https://github.com/aws/aws-sdk-ruby/blob/431c33756ae4c3a020a9aa26e3fce9ae0f8673d6/apis/monitoring/2010-08-01/api-2.json#L219-L235))
**Error Shape**: `DashboardNotFoundError` ([model def](https://github.com/aws/aws-sdk-ruby/blob/431c33756ae4c3a020a9aa26e3fce9ae0f8673d6/apis/monitoring/2010-08-01/api-2.json#L884-L895))
  *  Note that this error shape has a custom error code defined (`ResourceNotFound`)

We receive the following error response:
```
"<ErrorResponse xmlns=\"http://monitoring.amazonaws.com/doc/2010-08-01/\">\n  <Error>\n    <Type>Sender</Type>\n    <Code>ResourceNotFound</Code>\n    <Message>Dashboard DashboardName does not exist</Message>\n  </Error>\n  <RequestId>fooRequestID</RequestId>\n</ErrorResponse>\n"
```
which brings us to this raise:
```
Aws::CloudWatch::Errors::ResourceNotFound: Dashboard DashboardName does not exist
```
This is intended behavior from what I read on query protocol per Smithy Docs linked above